db.cpp

```diff
@@ -175,22 +175,31 @@ void CAddrDb::GetIPs_(set<CNetAddr>& ips, int max, const bool* nets) {
     } else {
       id = *ourId.begin();
     }
-    if (id >= 0) {
+    if (id >= 0 && idToInfo[id].services & requestedFlags) {
```

Comment:
No, I think it needs to be:
```
if (id >= 0 && (idToInfo[id].services & requestedFlags) == requestedFlags) {
```

Otherwise if you ask for flags 5, you may get entries with flags 1 or 4.